multiqc/modules/mirtrace/mirtrace.py

```diff
@@ -0,0 +1,381 @@
+            'id': 'mirtrace_length_plot',
+            'title': 'miRTrace: Read Length Distribution',
+            'ylab': 'Read Count',
+            'xlab': 'Read Lenth (bp)',
+            'ymin': 0,
+            'xmin': 0,
+            'xDecimals': False,
+            'tt_label': '<b>Read Length (bp) {point.x}</b>: {point.y} Read Count',
+            'xPlotBands': [
+                {'from': 40, 'to': 50, 'color': '#ffebd1'},
+                {'from': 26, 'to': 40, 'color': '#e2f5ff'},
+                {'from': 18, 'to': 26, 'color': '#e5fce0'},
+                {'from': 0, 'to': 18, 'color': '#ffffe2'},
+            ]
+        }
+
+        return linegraph.plot(data, config)
+
+
+    # miRTrace RNA Categories
+    def mirtrace_rna_categories(self):
+        """ Generate the miRTrace RNA Categories"""
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys['reads_mirna']      =   { 'color': '#33a02c', 'name': 'miRNA' }
+        keys['reads_rrna']       =   { 'color': '#ff7f00', 'name': 'rRNA' }
+        keys['reads_trna']       =   { 'color': '#1f78b4', 'name': 'tRNA' }
+        keys['reads_artifact']   =   { 'color': '#fb9a99', 'name': 'Artifact' }
+        keys['reads_unknown']    =   { 'color': '#d9d9d9', 'name': 'Unknown' }
+
+        # Construct a data structure for the plot - duplicate the samples for read 1 and read 2
+        data = {}
+        for s_name in self.summary_data:
+            data[s_name]                                =    {}
+            data[s_name]['reads_mirna']      =   self.summary_data[s_name]['reads_mirna']
+            data[s_name]['reads_rrna']       =   self.summary_data[s_name]['reads_rrna']
+            data[s_name]['reads_trna']       =   self.summary_data[s_name]['reads_trna']
+            data[s_name]['reads_artifact']   =   self.summary_data[s_name]['reads_artifact']
+            data[s_name]['reads_unknown']    =   self.summary_data[s_name]['reads_unknown']
+
+
+        # Config for the plot
+        config = {
+            'id': 'mirtrace_rna_categories_plot',
+            'title': 'miRTrace: RNA Categories',
+            'ylab': '# Reads',
+            'cpswitch_counts_label': 'Number of Reads'
+        }
+
+        return bargraph.plot(data, keys, config)
+
+
+    # miRTrace Contamination Check
+    def mirtrace_contamination_check(self):
+        """ Generate the miRTrace Contamination Check"""
+
+        # A library of 24 colors. Should be enough for this plot
+        color_lib = ['rgb(166,206,227)', 'rgb(31,120,180)', 'rgb(178,223,138)', 'rgb(51,160,44)', 'rgb(251,154,153)', 'rgb(227,26,28)', 'rgb(253,191,111)', 'rgb(255,127,0)', 'rgb(202,178,214)', 'rgb(106,61,154)', 'rgb(255,255,153)', 'rgb(177,89,40)', 'rgb(141,211,199)', 'rgb(255,255,179)', 'rgb(190,186,218)', 'rgb(251,128,114)', 'rgb(128,177,211)', 'rgb(253,180,98)', 'rgb(179,222,105)', 'rgb(252,205,229)', 'rgb(217,217,217)', 'rgb(188,128,189)', 'rgb(204,235,197)', 'rgb(255,237,111)']
```

Comment:
Any reason not to use the default colours here?

Reply:
Ah ok, makes sense. The default bar plotting colours are from HighCharts actually, not colorbrewer. But the same thing applies - only 10 defaults before cycling: https://api.highcharts.com/highcharts/colors

Fine to keep it as there's a reason 👍